modules/dists/SparseBlockDist.chpl

```diff
@@ -117,6 +116,64 @@ class SparseBlockDom: BaseSparseDom {
     dsiAdd((ind,));
   }
 
+  proc dsiBulkAdd(inds: [] index(rank, idxType), isSorted=false, isUnique=false,
+      preserveInds=true){
+  
+    if !isSorted && preserveInds {
+      var _inds = inds;
+      bulkAdd_help(_inds, isSorted, isUnique); 
+    }
+    else {
+      bulkAdd_help(inds, isSorted, isUnique);
+    }
+  }
+
+  // Tried to put this record in the function and the if statement, but got a
+  // segfault from the compiler.
```

Comment:
That would be a nice addition. Maybe we can even support larger set of index comparators. I can already think of ways they might be useful(eg partial reductions, 1-d iterators of multidimensional data etc). They can be used to sort indices in row- or column-major order. Such support can even be in DSIUtil maybe, or at least a bit more decoupled from bulkAdd?